spring-integration-core/src/main/java/org/springframework/integration/gateway/GatewayProxyFactoryBean.java

```diff
@@ -549,10 +550,17 @@ else if (this.methodMetadataMap != null && this.methodMetadataMap.size() > 0) {
 				}
 			}
 		}
+		Map<String, Object> headers = null;
+		// We don't want to eagerly resolve the error channel here
+		Object ec = this.errorChannel == null ? this.errorChannelName : this.errorChannel;
```

Comment:
I think `errorChannel` local variable would be good enough.
Yes, just nit-picking, but still Open Source, so good names in good code makes us proud :smile: